cfsdk/src/main/java/io/harness/cfsdk/cloud/repository/FeatureRepositoryImpl.java

```diff
@@ -42,22 +42,25 @@ public Evaluation getEvaluation(
             final String evaluationId,
             final String cluster
     ) {
-        if (networkInfoProvider.isNetworkAvailable()) {
 
-            ApiResponse apiResponse = this.featureService.getEvaluationForId(
+        Evaluation eval = cloudCache.getEvaluation(buildKey(environment, target), evaluationId);
+
+        if (eval == null && networkInfoProvider.isNetworkAvailable()) {
```

Comment:
Suggestion: is it worth adding a clause for

`eval == null && !networkInfoProvider.isNetworkAvailable` or is it the responsibility of callers to check for a `null` evaluation?

Reply:
Oops. Ignore. I can see the test case to check for `null` when no cache or network! :)

Reply:
That's right. The getEvaluationById function in CfClient will check for the null and return the default value before returning that to the xVariation method.